### PR TITLE
Add a flag to control default dispatch policy per zone.

### PR DESCRIPTION
**Pull request by baldersheim**
@hakonhall and @hmusum PR

### CODE REVIEW

File: config-model/src/test/java/com/yahoo/config/model/MockModelContext.java [hunk 1]
```diff
@@ -139,6 +139,9 @@ public Zone zone() {
 
             @Override
             public boolean useFdispatchByDefault() { return true; }
+
+            @Override
+            public boolean useAdaptiveDispatch() { return false; }
```
Comment (hakonhall) [inline config-model/src/test/java/com/yahoo/config/model/MockModelContext.java]: This Properties implementation can probably be replaced by TestProperties...

File: configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java [hunk 2]
```diff
@@ -158,6 +159,8 @@ public Properties(ApplicationId applicationId,
                     .with(FetchVector.Dimension.APPLICATION_ID, applicationId.serializedForm()).value();
             this.useFdispatchByDefault = Flags.USE_FDISPATCH_BY_DEFAULT.bindTo(flagSource)
                     .with(FetchVector.Dimension.APPLICATION_ID, applicationId.serializedForm()).value();
+            this.useAdaptiveDispatch = Flags.USE_ADAPTIVE_DISPATCH.bindTo(flagSource)
+                    .with(FetchVector.Dimension.ZONE_ID, zone.toString()).value();
```
Comment (hakonhall) [inline configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java]: ZONE_ID is not what you think it is. All flags are automatically per zone. The ZONE_ID dimension is only useful if the controller wants to resolve the flag to different values depending on which zone it is operating on.

Consider adding the APPLICATION_ID dimension to allow you to control this per application (and update the Flags definition accordingly).

**Review by hmusum — APPROVED** [review]
LGTM

File: configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java (hunk 2 above)
Comment (baldersheim) [reply configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java]: Done, but I hope I can control for the whole zone too......

File: config-model/src/test/java/com/yahoo/config/model/MockModelContext.java (hunk 1 above)
Comment (baldersheim) [reply config-model/src/test/java/com/yahoo/config/model/MockModelContext.java]: Done

File: configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java (hunk 2 above)
Comment (hakonhall) [reply configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java]: yes

**Review by hakonhall — APPROVED** [review]

**Pull request merged**